radix-engine/src/wasm/prepare.rs

```diff
@@ -161,6 +163,32 @@ impl WasmModule {
         Ok(self)
     }
 
+    pub fn check_exports(
+        self,
+        blueprints: &HashMap<String, BlueprintAbi>,
+    ) -> Result<Self, PrepareError> {
+        // only allow `env::radix_engine` import
+        let exports = self
+            .module
+            .export_section()
+            .ok_or(PrepareError::NoExports)?;
+        for (_, blueprint_abi) in blueprints {
+            for func in &blueprint_abi.fns {
+                let func_name = &func.export_name;
+                // TODO: Check if signature matches
+                if !exports.entries().iter().any(|x| {
+                    x.field().eq(func_name) && matches!(x.internal(), Internal::Function(_))
+                }) {
+                    return Err(PrepareError::MissingExport {
+                        export_name: func_name.to_string(),
+                    });
+                }
+            }
+        }
+
+        Ok(self)
```

Comment:
Should we reject exports that's not specified in the ABI? :)

Reply:
hrm do you think that gives us more security? There may be interesting usecases if we allowed other exports?

Reply:
Not really, just more control over the structure of the wasm. Let's keep it as-in for now.